data/available_versions_test.go

```diff
@@ -13,6 +14,69 @@ import (
 	"github.com/deis/workflow-manager/pkg/swagger/models"
 )
 
+// Creating a novel mock struct that fulfills the AvailableVersions interface
+type testAvailableVersions struct{}
+
+func (a testAvailableVersions) Refresh(cluster models.Cluster) ([]models.ComponentVersion, error) {
+	data := getMockComponentVersions()
+	var componentVersions []models.ComponentVersion
+	_ = json.Unmarshal(data, &componentVersions)
+	return componentVersions, nil
+}
+
+func (a testAvailableVersions) Store(c []models.ComponentVersion) {
+	return
+}
+
+func (a testAvailableVersions) Cached() []models.ComponentVersion {
+	return nil
+}
+
+// Creating another mock struct that fulfills the AvailableVersions interface
+type shouldBypassAvailableVersions struct{}
+
+func (a shouldBypassAvailableVersions) Refresh(cluster models.Cluster) ([]models.ComponentVersion, error) {
+	var componentVersions []models.ComponentVersion
+	data := []byte(fmt.Sprintf(`[{
+	  "components": [
+	    {
+	      "component": {
+	        "name": "bypass me",
+	        "description": "bypass me"
+	      },
+	      "version": {
+	        "version": "v2-bypass"
+	      }
+	    }
+	  ]
+	}]`))
+	_ = json.Unmarshal(data, &componentVersions)
```

Comment:
you should either return this error or just construct the `[]models.ComponentVersion` directly and return it